services/harmony/app/frontends/ogc-edr/get-data-for-trajectory.ts

```diff
@@ -0,0 +1,233 @@
+*
+* @param wktMultiLineString - The WKT MULTILINESTRING string to convert.
+* @param sideLength - The buffer distance (side length) to create around each LineString.
+* @returns The converted WKT MULTIPOLYGON string.
+* @throws RequestValidationError if the WKT MULTILINESTRING string format is invalid.
+*/
+function wktMultiLineStringToMultipolygon(
+  wktMultiLineString: string,
+  sideLength: number): string {
+
+  validateWkt(wktMultiLineString);
+  const match = wktMultiLineString.match(/MULTILINESTRING\s*\(\((.*)\)\)/);
+  if (!match) {
+    throw new RequestValidationError(
+      `query parameter "coords" invalid WKT MULTILINESTRING format: ${wktMultiLineString}`);
+  }
+
+  const multiLS = match[1].replace(/\),\s*\(/g, '),(');
+
+  const lineStrings = multiLS.split('),(').map(lineStr => lineStr.trim());
+
+  let polygons: string[] = [];
+
+  // Convert each LineString to MultiPolygon and add to result
+  lineStrings.forEach(lineString => {
+    const linePolygons = convertLineStringCoordsToPolygons(lineString, sideLength);
+    polygons = polygons.concat(linePolygons);
+  });
+
+  return `MULTIPOLYGON (${polygons.join(', ')})`;
+}
+
+/**
+* Converts a WKT LINESTRING/MULTILINESTRING string to a WKT POLYGON/MULTIPOLYGON string.
+*
+* @param wkt - The WKT LINESTRING or WKT MULTILINESTRING string to convert.
+* @param sideLength - The length of the side around each endpoint of the line,
+*                     defaults to 0.0001. It is about 11 meters in precision.
+* @returns The converted WKT POLYGON or WKT MULTIPOLYGON string.
+* @throws RequestValidationError if the WKT string format is invalid.
+*/
+export function convertWktLineToPolygon(
+  wkt: string,
+  sideLength: number = LINESTRING_PRECISION): string {
+
+  if (wkt.startsWith('LINESTRING')) {
+    return convertLineStringToPolygon(wkt, sideLength);
+  } else if (wkt.startsWith('MULTILINESTRING')) {
+    return wktMultiLineStringToMultipolygon(wkt, sideLength);
+  } else {
+    throw new RequestValidationError(`query parameter "coords" invalid WKT format: ${wkt}`);
+  }
+}
+
+/**
+ * Express middleware that responds to OGC API - EDR Trajectory GET requests.
+ * Responds with the actual EDR data.
+ *
+ * @param req - The request sent by the client
+ * @param res - The response to send to the client
+ * @param next - The next express handler
+ */
+export function getDataForTrajectory(
+  req: HarmonyRequest,
+  res: Response,
+  next: NextFunction,
+): void {
+
+  getDataCommon(req);
+  const query = keysToLowerCase(req.query);
+  const { operation } = req;
+
+  if (query.coords) {
+    const polygon = convertWktLineToPolygon(query.coords);
+    try {
+      const geoJson = parseWkt(polygon);
+      if (geoJson) {
+        operation.geojson = JSON.stringify(geoJson);
+      }
+    } catch (e) {
+      if (e instanceof ParameterParseError) {
+        // Turn parsing exceptions into 400 errors pinpointing the source parameter
```

Comment:
If we're throwing a server error it's a 500 error. If it's a parsing issue I think we want to throw a RequestValidationError which results in a 400 rather than a 500.

Reply:
I will update it to RequestValidationError.